zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/OriginalDataAgent.scala

```diff
@@ -2,29 +2,39 @@ package edu.uci.ics.cloudberry.zion.actor
 
 import akka.actor.Props
 import akka.pattern.pipe
-import edu.uci.ics.cloudberry.zion.actor.OriginalDataAgent.Cardinality
 import edu.uci.ics.cloudberry.zion.common.Config
 import edu.uci.ics.cloudberry.zion.model.datastore.{IDataConn, IQLGenerator}
+import edu.uci.ics.cloudberry.zion.model.impl.DataSetInfo
 import edu.uci.ics.cloudberry.zion.model.schema._
 import org.joda.time.{DateTime, Duration}
 import play.api.libs.json._
 
 import scala.concurrent.ExecutionContext
 
-class OriginalDataAgent(override val dbName: String,
-                        override val schema: Schema,
-                        val initCardinality: Cardinality,
+class OriginalDataAgent(val dataSetInfo: DataSetInfo,
                         override val queryParser: IQLGenerator,
                         override val conn: IDataConn,
                         override val config: Config)(implicit ec: ExecutionContext)
-  extends AbstractDataSetAgent(dbName, schema, queryParser, conn, config)(ec) {
+  extends AbstractDataSetAgent(dataSetInfo.name, dataSetInfo.schema, queryParser, conn, config)(ec) {
 
   import OriginalDataAgent._
 
-  val lastCount: Cardinality = initCardinality
+  val lastCount: Cardinality = new Cardinality(
+    dataSetInfo.dataInterval.getStart,
+    dataSetInfo.dataInterval.getEnd,
+    dataSetInfo.stats.cardinality
+  )
 
-  //update Stats periodically
-  context.system.scheduler.schedule(config.AgentCollectStatsInterval, config.AgentCollectStatsInterval, self, UpdateStats)
+  /**
+    * When the Agent starts,
+    *   1. Ask for the stats from maxTimeStamp till now,
+    *   2. Start a scheduler to query for cardinality periodically.
+    * Stats including: minTimeStamp, maxTimeStamp, cardinality.
+    */
+  override def preStart(): Unit = {
+    collectStats(lastCount.till)
+    context.system.scheduler.schedule(config.AgentCollectStatsInterval, config.AgentCollectStatsInterval, self, UpdateStats)
```

Comment:
I think you can merge this two as 
`context.system.scheduler.schedule(0, config.AgentCollectStatsInterval, self, UpdateStats)`